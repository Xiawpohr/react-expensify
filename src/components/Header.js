import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { logout } from '../actions/auth.js'

export const Header = (props) => (
  <header>
    <h1>Expensify APP</h1>
    <NavLink to='/' activeClassName='active-link' exact>DashBoard</NavLink>
    <NavLink to='/create' activeClassName='active-link' >Create Expense</NavLink>
    <NavLink to='/help' activeClassName='active-link' >Help</NavLink>
    <button onClick={props.logout} >Log Out</button>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout())
})

export default connect(undefined, mapDispatchToProps)(Header)
