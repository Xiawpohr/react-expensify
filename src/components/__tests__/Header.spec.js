import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../Header.js'

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
})
