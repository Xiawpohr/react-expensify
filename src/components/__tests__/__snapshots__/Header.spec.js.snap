// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify APP
  </h1>
  <NavLink
    activeClassName="active-link"
    aria-current="page"
    exact={true}
    to="/dashboard"
  >
    DashBoard
  </NavLink>
  <NavLink
    activeClassName="active-link"
    aria-current="page"
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="active-link"
    aria-current="page"
    to="/help"
  >
    Help
  </NavLink>
  <button
    onClick={[Function]}
  >
    Log Out
  </button>
</header>
`;
